daliuge-engine/dlg/drop.py

```diff
@@ -427,28 +427,26 @@ def get_param_value(attr_name, default_value):
             has_app_param = (
                     "applicationArgs" in kwargs and attr_name in kwargs["applicationArgs"]
             )
-            param = default_value
-            if has_component_param and has_app_param:
-                logger.warning(
-                    "Drop has both component and app param %s. Using component param.",
-                    attr_name
-                )
+            # c_param = a_param = default_value
             if has_component_param:
-                param = kwargs.get(attr_name)
+                c_param = kwargs.get(attr_name)
             elif has_app_param:
-                if kwargs["applicationArgs"].get(attr_name).usage in [
+                if kwargs["applicationArgs"].get(attr_name).usage not in [
                     "InputPort",
                     "OutputPort",
                     "InputOutput",
                 ]:
-                    # inp = kwargs["input"]
-                    # param = pickle.loads(
-                    #     droputils.allDropContents()
-                    #     )
-                    pass
-                else:
-                    param = kwargs["applicationArgs"].get(attr_name).value
-            return param
+                    a_param = kwargs["applicationArgs"].get(attr_name).value
+            if has_component_param:
+                # logger.warning(
```

Comment:
There's quite a few comments that exist that we can delete now that testing is done.

Reply:
This is the second check for `has_component_param`. We don't need the variables `c_param`, `a_param`, we should be able to return `kwargs.get(attr_name)` immediately.